tools/rosout/rosout.cpp

```diff
@@ -120,6 +126,11 @@ class Rosout
   {
     agg_pub_.publish(msg);
 
+    if (!handle_)
+    {
+      return;
+    }
```

Comment:
It'd be nice if this could handled as a RAII thing when it goes out of scope, but appreciating that you're just trying to patch in a small fix and not rewrite the whole thing, LGTM.